Sources/Datadog/Core/System/LaunchTimeProvider.swift

```diff
@@ -0,0 +1,48 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import UIKit
+import _Datadog_Private
+
+/// Provides the application launch time.
+internal protocol LaunchTimeProviderType {
+    /// The app process launch duration (in seconds) measured as the time from loading the first SDK object into memory
+    /// to receiving `UIApplication.didBecomeActiveNotification` notification.
+    var launchTime: TimeInterval? { get }
+}
+
+internal class LaunchTimeProvider: LaunchTimeProviderType {
+    private let notificationCenter: NotificationCenter
+    private let queue = DispatchQueue(
+        label: "com.datadoghq.launch-time-provider", qos: .userInteractive
+    )
+
+    // MARK: - Initialization
+
+    init(notificationCenter: NotificationCenter = .default) {
+        self.notificationCenter = notificationCenter
+        notificationCenter.addObserver(
+            self, selector: #selector(stopLaunchTimer), name: UIApplication.didBecomeActiveNotification, object: nil
```

Comment:
it could be nice to warn the user if they `initialize` too late, after `didBecomeActiveNotification` was fired
i think we can catch that by checking `UIApplication.shared.state` somewhere around here

Reply:
It's a good point 👍. However, I did more than just the warning. The new push makes the launch time collection independent from SDK initialization, so the users are not limited to calling `Datadog.initialize()` before the app becomes active. Please review 🙏 👀.